projects/angular-auth-oidc-client/src/lib/authState/auth-state.service.ts

```diff
@@ -100,11 +100,14 @@ export class AuthStateService {
 
         this.loggerService.logDebug(`authorizedState in storage is ${currentAuthState}`);
 
-        if (this.tokenIsExpired()) {
-            this.loggerService.logDebug('persisted token is expired');
+        if (this.hasIdTokenExpired()) {
+            this.loggerService.logDebug('persisted id_token is expired');
+            return false;
+        } else if (this.hasAccessTokenExpiredIfExpiryExists()) {
+            this.loggerService.logDebug('persisted access_token is expired');
             return false;
         } else {
-            this.loggerService.logDebug('persisted token is valid');
+            this.loggerService.logDebug('persisted id_token and access token are valid');
```

Comment:
remove the `else` keyword for clarity. As we work with early returns we do not need the else.

Reply:
fixed